inference-engine/src/cldnn_engine/ops/loop.cpp

```diff
@@ -0,0 +1,250 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+#include "cldnn_program.h"
+#include "cldnn_common_utils.h"
+#include "cldnn_engine.h"
+
+#include <cpp/ie_cnn_network.h>
+
+#include "ngraph/op/loop.hpp"
+#include "ngraph/op/constant.hpp"
+#include "ngraph/op/util/sub_graph_base.hpp"
+#include "transformations/utils/utils.hpp"
+
+#include "cldnn/primitives/loop.hpp"
+#include "cldnn/primitives/mutable_data.hpp"
+#include "cldnn/primitives/data.hpp"
+#include "cldnn/primitives/reorder.hpp"
+#include "cldnn/graph/topology.hpp"
+
+#include <vector>
+#include <algorithm>
+
+using Loop = ngraph::op::v5::Loop;
+
+namespace CLDNNPlugin {
+
+template<class DATA_TYPE>
+static DATA_TYPE CreateScalarData(Program &p, const cldnn::primitive_id& id, int64_t num) {
+    auto mem = p.GetEngine().allocate_memory({ cldnn::data_types::i64, cldnn::format::bfyx, { 1, 1, 1, 1 } });
+    cldnn::mem_lock<int64_t> ptr{mem, p.GetEngine().get_program_stream()};
+    *ptr.begin() = num;
+    return {id, mem};
+}
+
+static cldnn::mutable_data CreateAdditionalOutputData(Program &p, const std::shared_ptr<ngraph::Node>& op,
+                                                        const cldnn::primitive_id& id, const cldnn::primitive_id& input,
+                                                        const int32_t output_idx) {
+    const auto precision = DataTypeFromPrecision(op->get_output_element_type(output_idx));
+    const auto format = DefaultFormatForDims(op->get_output_shape(output_idx).size());
+    const auto tensor = CldnnTensorFromIEDims(op->get_output_shape(output_idx));
+    cldnn::layout output_layout = cldnn::layout(precision, format, tensor);
+    auto mem = p.GetEngine().allocate_memory(output_layout);
+    auto md = cldnn::mutable_data(id, {input}, mem); // cldnn::data cannot set dependency
+    return md;
+}
+
+static void UpdateBackedge(std::vector<cldnn::loop::backedge_mapping>& back_edges,
+                            const cldnn::primitive_id& old_primitive_id, const cldnn::primitive_id& new_primitive_id) {
+    for (auto& back_edge : back_edges) {
+        if (back_edge.from == old_primitive_id) {
+            back_edge.from = new_primitive_id;
+        }
+    }
+}
+
+static std::string GetExternalInputName(const int64_t body_parameter_index,
+                                        const std::shared_ptr<Loop>& op) {
+    const auto& loop_input_descs = op->get_input_descriptions();
+    for (const auto& loop_input_desc : loop_input_descs) {
+        if (loop_input_desc->m_body_parameter_index == body_parameter_index) {
+            auto external_node = op->get_input_node_shared_ptr(loop_input_desc->m_input_index);
+            return layer_type_name_ID(external_node);
+        }
+    }
+    return {""};
+}
+
+static InferenceEngine::Precision GetPrecisionFromNgraphType(ngraph::element::Type_t type) {
```

Comment:
Why don't you use `convertPrecision` function from `inference-engine/src/plugin_api/ie_ngraph_utils.hpp`?

Reply:
Fixed it. thanks.